mnemosyne/core/filters.py

```
@@ -0,0 +1,346 @@
+_compiled_noise: Optional[List[re.Pattern]] = None
+_compiled_secrets: Optional[List[re.Pattern]] = None
+
+
+def _compile_patterns(patterns: List[str]) -> List[re.Pattern]:
+    """Compile a list of regex strings, skipping invalid ones."""
+    compiled = []
+    for p in patterns:
+        try:
+            compiled.append(re.compile(p, re.IGNORECASE))
+        except re.error:
+            logger.debug("Invalid pattern, skipping: %r", p)
```

Ruff (0.15.20)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)